site/content/how-to/traffic-management/response-headers.md

```diff
@@ -0,0 +1,155 @@
+   ```
+
+Verify if the Pod is running in the `default` Namespace:
+
+   ```shell
+   kubectl -n default get pods
+   ```
+
+   ```text
+   NAME                      READY   STATUS    RESTARTS   AGE
+   headers-6f854c478-k9z2f   1/1     Running   0          32m
+   ```
+
+### Deploy the Gateway API Resources for the Coffee Applications
+
+
+The [gateway](https://gateway-api.sigs.k8s.io/api-types/gateway/) resource is typically deployed by the [cluster operator](https://gateway-api.sigs.k8s.io/concepts/roles-and-personas/#roles-and-personas_1). To deploy the gateway:
+
+```yaml
+kubectl apply -f - <<EOF
+apiVersion: gateway.networking.k8s.io/v1
+kind: Gateway
+metadata:
+  name: gateway
+spec:
+  gatewayClassName: nginx
+  listeners:
+  - name: http
+    port: 80
+    protocol: HTTP
+EOF
+```
+
+### Configure Routing
+
+This gateway defines a single listener on port 80. Since no hostname is specified, this listener matches on all hostnames.
+
+The [HTTPRoute](https://gateway-api.sigs.k8s.io/api-types/httproute/) is typically deployed by the [application developer](https://gateway-api.sigs.k8s.io/concepts/roles-and-personas/#roles-and-personas_1). To deploy the `headers` HTTPRoute:
+
+```yaml
+kubectl apply -f - <<EOF
+apiVersion: gateway.networking.k8s.io/v1
+kind: HTTPRoute
+metadata:
+  name: headers
+spec:
+  parentRefs:
+  - name: gateway
+    sectionName: http
+  hostnames:
+  - "cafe.example.com"
+  rules:
+  - matches:
+    - path:
+        type: PathPrefix
+        value: /headers
+    filters:
+    - type: ResponseHeaderModifier
+      responseHeaderModifier:
+        set:
+        - name: Response-Overwrite-Header
+          value: overwritten-value
+        add:
+        - name: My-cool-header
+          value: this-is-the-value
+        remove:
+        - X-Server-Version
+    backendRefs:
+    - name: headers
+      port: 80
+EOF
+```
+
+This HTTPRoute has a few important properties:
+
+- The `parentRefs` references the gateway resource that we created, and specifically defines the `http` listener to attach to, via the `sectionName` field.
+- `cafe.example.com` is the hostname that is matched for all requests to the backends defined in this HTTPRoute.
+- The `match` rule defines that all requests with the path prefix `/headers` are sent to the `headers` Service.
+- There is `ResponseHeaderModifier` filter defined for the path prefix `/headers` to overwrite value for header `Response-Overwrite-Header`, append value to header `My-cool-header` and remove `X-Server-Version` header from the HTTP response.
+
+
+### Send Traffic to Headers
+
+To access the application, we will use `curl` to send requests to the `headers` endpoint.
+
+Notice our configured header values can be seen in the `responseHeaders` section below, and that the `X-Server-Version` header is absent. The header `My-cool-header` gets appended with the new value and `Response-Overwrite-Header` gets overwritten to `overwritten-value` as defined in the *HttpRoute*.
```

Comment:
Before this sentence, let's list the response headers that the headers application normally returns so it's clear what the filter is doing.


Or actually, I think it would be more powerful if we did the following:
- Create an HTTPRoute without the filter
- Send a request with `-i` and show the response header output
- Update the HTTPRoute and add the filter
- Send another request and show how the headers have been modified.

Reply:
i have updated the tutorial as described above
1. Create a basic HTTPRoute, record results.
2. Create a filter HTTPRoute, record results.